### PR TITLE
Date picker redesign

### PR DESCRIPTION
**Pull request by savelichalex**

### CODE REVIEW

File: packages/flask/src/UIDateTimePicker/Calendar/useCalendar.ts [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+    selectedYear: number,
+    dayNamesShort: string[],
+    min?: Date,
+    max?: Date,
+) {
+    const date = dayjs().month(selectedMonth).year(selectedYear);
+    /**
+     * TODO: it's for now starts with Sunday
+     *       this is wrong and should be localized!
+     */
```
Comment (sertony) [inline packages/flask/src/UIDateTimePicker/Calendar/useCalendar.ts:50]: True

File: packages/flask/src/UIDateTimePicker/Calendar/useCalendar.ts [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+}
+
+export function useDaysCalendar() {
+    const {
+        state: { selectedDate, selectedMonth, selectedYear },
+        dispatch,
+        min,
+        max,
+    } = useDateTimeState();
+    // TODO: it should be last valid rather!!!
```
Comment (sertony) [inline packages/flask/src/UIDateTimePicker/Calendar/useCalendar.ts]: Pointing in order not to forget

File: packages/flask/src/UIDateTimePicker/Time/Time.tsx [hunk 3]
```diff
@@ -190,6 +181,9 @@ export function UITimeInput() {
             : inputRef.current?.focus();
     }, []);
 
+    // TODO: move ScrollView to DatePicker to cover date/time mode
+    // TODO: don't forget to set maxHeight for it, to cover small screens
+    //       when date/time won't fit with keyboard open!
```
Comment (sertony) [inline packages/flask/src/UIDateTimePicker/Time/Time.tsx:186]: Please, don't forget =)

File: packages/flask/src/UIDateTimePicker/useDateTimeState.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    mode: UIDateTimePickerMode;
+    state: DateTimeState;
+    isTimeValid: boolean;
+    dispatch: React.Dispatch<DateTimeAction>;
+    min?: Date;
+    max?: Date;
+    isAmPmTime?: boolean;
+};
+
+// TODO: change type!!!
```
Comment (sertony) [inline packages/flask/src/UIDateTimePicker/useDateTimeState.tsx]: one more thing

**Review by sertony — APPROVED** [review]
Approved, but there are some TODOs left, besides am I right that it doesn't contain any breaking changes?

File: packages/flask/src/UIDateTimePicker/Time/Time.tsx (hunk 3 above)
Comment (savelichalex) [reply packages/flask/src/UIDateTimePicker/Time/Time.tsx:186]: Have a separate task for it :)

**Comment by savelichalex** [comment]
> Approved, but there are some TODOs left, besides am I right that it doesn't contain any breaking changes?

Yeah, I guess it shouldn't break. Besides, AFAIK it wasn't used yet :D

**Pull request merged**